crates/cheatnet/tests/cheatcodes/set_balance.rs

```diff
@@ -0,0 +1,125 @@
+use cheatnet::runtime_extensions::forge_runtime_extension::cheatcodes::erc20::Token;
+use cheatnet::runtime_extensions::forge_runtime_extension::cheatcodes::erc20::set_balance;
+use conversions::felt::FromShortString;
+use conversions::string::TryFromHexStr;
+use starknet::core::types::U256;
+use starknet::core::utils::get_selector_from_name;
+use starknet_api::contract_class::SierraVersion;
+use starknet_api::core::ClassHash;
+use starknet_api::core::ContractAddress;
+use starknet_types_core::felt::Felt;
+
+trait SetBalanceTrait {
+    fn set_balance(&mut self, target: ContractAddress, new_balance: U256, token: Token);
+}
+
+impl SetBalanceTrait for TestEnvironment {
+    fn set_balance(&mut self, target: ContractAddress, new_balance: U256, token: Token) {
+        set_balance(&mut self.cached_state, target, new_balance, token).unwrap();
+    }
+}
+
+fn get_balance(
+    test_env: &mut TestEnvironment,
+    target: ContractAddress,
+    token: Token,
+) -> CallResult {
+    test_env.call_contract(&token.contract_address(), "balance_of", &[target.into()])
+}
+
+fn declare_and_deploy_token_strk(test_env: &mut TestEnvironment) {
+    let class_hash = ClassHash::try_from_hex_str(STRK_CLASS_HASH).unwrap();
+    let contract_class = RunnableCompiledClass::V1(
+        CompiledClassV1::try_from_json_string(STRK_ERC20_CASM, SierraVersion::LATEST).unwrap(),
+    );
+    declare_with_contract_class(&mut test_env.cached_state, contract_class, class_hash)
+        .expect("Failed to declare STRK token");
+
+    deploy_token_strk(test_env);
+}
+
+fn deploy_token_strk(test_env: &mut TestEnvironment) {
+    let class_hash = ClassHash::try_from_hex_str(STRK_CLASS_HASH).unwrap();
+    let contract_address = ContractAddress::try_from_hex_str(STRK_CONTRACT_ADDRESS).unwrap();
+    let constructor_calldata = strk_constructor_calldata();
+    deploy_at_wrapper(
+        &mut test_env.cached_state,
+        &mut test_env.cheatnet_state,
+        &class_hash,
+        &constructor_calldata,
+        contract_address,
+    )
+    .expect("Failed to deploy STRK token");
+}
+
+#[test]
+fn test_set_balance_strk() {
+    let mut test_env = TestEnvironment::new();
+
+    declare_and_deploy_token_strk(&mut test_env);
+
+    let contract_address = ContractAddress::try_from_hex_str("0x123").unwrap();
+
+    let token = Token::STRK;
+    let balance = get_balance(&mut test_env, contract_address, token);
+    assert_success(balance, &[0.into(), 0.into()]);
+
+    test_env.set_balance(contract_address, U256::from(1_000_000_u32), token);
+    let balance = get_balance(&mut test_env, contract_address, token);
+    assert_success(balance, &[1_000_000.into(), 0.into()]);
+
+    test_env.set_balance(contract_address, U256::from_words(u128::MAX, 100), token);
+    let balance = get_balance(&mut test_env, contract_address, token);
+    assert_success(balance, &[u128::MAX.into(), 100.into()]);
+}
```

Comment:
note: We do deployment here because the function running test case (which includes predeployment) isn't executed in cheatnet env.